modules/core/resources/META-INF/smithy/test/pizza.smithy

```diff
@@ -0,0 +1,324 @@
+    @httpQuery("query")
+    query: String,
+    body: String
+}
+
+structure HeaderEndpointData {
+    @httpHeader("X-UPPERCASE-HEADER")
+    uppercaseHeader: String,
+    @httpHeader("X-Capitalized-Header")
+    capitalizedHeader: String,
+    @httpHeader("x-lowercase-header")
+    lowercaseHeader: String,
+    @httpHeader("x-MiXeD-hEaDEr")
+    mixedHeader: String,
+}
+
+structure AddMenuItemResult {
+    @httpPayload
+    @required
+    itemId: String,
+    @timestampFormat("epoch-seconds")
+    @httpHeader("X-ADDED-AT")
+    @required
+    added: Timestamp
+}
+
+
+
+
+structure VersionOutput {
+    @httpPayload
+    @required
+    version: String
+}
+
+@error("client")
+structure PriceError {
+    @required
+    message: String,
+    @required
+    @httpHeader("X-CODE")
+    code: Integer
+}
+
+structure GetMenuRequest {
+    @httpLabel
+    @required
+    restaurant: String
+}
+
+structure GetMenuResult {
+    @required
+    @httpPayload
+    menu: Menu
+}
+
+@error("client")
+@httpError(404)
+structure NotFoundError {
+    @required
+    name: String
+}
+
+@error("client")
+structure FallbackError {
+    @required
+    error: String
+}
+
+map Menu {
+    key: String,
+    value: MenuItem
+}
+
+structure AddMenuItemRequest {
+    @httpLabel
+    @required
+    restaurant: String,
+    @httpPayload
+    @required
+    menuItem: MenuItem
+}
+
+structure MenuItem {
+    @required
+    food: Food,
+    @required
+    price: Float
+}
+
+union Food {
+    pizza: Pizza,
+    salad: Salad
+}
+
+structure Salad {
+    @required
+    name: String,
+    @required
+    ingredients: Ingredients
+}
+
+structure Pizza {
+    @required
+    name: String,
+    @required
+    base: PizzaBase,
+    @required
+    toppings: Ingredients
+}
+
+@enum([{name: "CREAM", value: "C"}, {name: "TOMATO", value: "T"}])
+string PizzaBase
+
+@enum([{value: "Mushroom"}, {value: "Cheese"}, {value: "Salad"}, {value: "Tomato"}])
+string Ingredient
+
+list Ingredients {
+    member: Ingredient
+}
+
+@error("server")
+@httpError(502)
+structure GenericServerError {
+    @required
+    message: String
+}
+
+@error("client")
+@httpError(418)
+structure GenericClientError {
+    @required
+    message: String
+}
+
+@readonly
+@http(method: "GET", uri: "/health", code: 200)
+operation Health {
+    input: HealthRequest,
+    output: HealthResponse,
+    errors: [ UnknownServerError ]
+}
+
+structure HealthRequest {
+    @httpQuery("query")
+    @length(min: 0, max: 5)
+    query: String
+}
+
+@freeForm(i : 1, a: 2)
+structure HealthResponse {
+    @required
+    status: String
+}
+
+// This error indicates a fatal, unexpected error has occurred. Fallback strategies ought to be triggered by this
+// error.
+@error("server")
+@httpError(500)
+structure UnknownServerError {
+    @required
+    errorCode: UnknownServerErrorCode,
+
+    description: String,
+
+    stateHash: String
+}
+
+// Define the singular error code that can be returned for an UnknownServerError
+@enum([
```

Comment:
why not use a 2.x enum?

Reply:
i should , i just copied this over from smithy4s